### PR TITLE
use @adobe/exc-app module

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
- replace custom implementation for loading the exc runtime object by the new @adobe/exc-app module
- + tiny bug fix for ui template: clear error message
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: generators/add-web-assets/exc-react/templates/src/App.js [hunk 1]
```diff
@@ -41,6 +41,25 @@ export default class App extends React.Component {
...
+      // }
+      // this.setState({ imsOrg, imsToken, locale })
+    })
+
+    // respond to history change events
+    this.props.runtime.on('history', ({ type, path }) => {
+      console.log('history change', { type, path })
+      const cleanedPath = path[0] === '/' ? path : '/' + path
+      if (type === 'external' && this.props.runtime.history.location.pathname !== cleanedPath) {
+        // this.props.runtime.history.replace(cleanedPath)
```
Comment (shazron) [inline generators/add-web-assets/exc-react/templates/src/App.js:59]: delete commented out code?

File: generators/add-web-assets/exc-react/templates/src/App.js [hunk 2]
```diff
@@ -41,6 +41,25 @@ export default class App extends React.Component {
 
     console.log('runtime object:', this.props.runtime)
     console.log('ims object:', this.props.ims)
+
+    // respond to configuration change events (e.g. user switches org)
+    this.props.runtime.on('configuration', ({ imsOrg, imsToken, locale }) => {
+      console.log('configuration change', { imsOrg, imsToken, locale })
+      // if (imsOrg !== this.state.imsOrg) {
```
Comment (shazron) [inline generators/add-web-assets/exc-react/templates/src/App.js:48]: delete commented out code?

Comment (shazron) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: the purpose of this handler now is just logging?

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 1 above)
Comment (shazron) [reply generators/add-web-assets/exc-react/templates/src/App.js:59]: purpose of this conditional now that the comments are commented out?

File: generators/add-web-assets/exc-react/templates/src/App.js [hunk 3]
```diff
@@ -63,7 +82,7 @@ export default class App extends React.Component {
       // invoke backend action
       const response = await actionWebInvoke(action, headers, params)
       // store the response
-      this.setState({ response })
+      this.setState({ response, errorMessage: null })
```
Comment (shazron) [inline generators/add-web-assets/exc-react/templates/src/App.js:85]: shouldn't `errorMessage` be `errorMsg`?

File: generators/add-web-assets/exc-react/templates/src/utils.js [hunk 4]
```diff
@@ -15,6 +15,19 @@ import actions from './config.json'
...
+ * the Experience Cloud Shell.
+ *
+ */
+function loadExcRuntime () {
+  /* eslint-disable-next-line */
+  (function(e,t){if(t.location===t.parent.location)throw new Error("Module Runtime: Needs to be within an iframe!");var o=function(e){var t=new URL(e.location.href).searchParams.get("_mr");return t||!e.EXC_US_HMR?t:e.sessionStorage.getItem("unifiedShellMRScript")}(t);if(!o)throw new Error("Module Runtime: Missing script!");if("https:"!==(o=new URL(decodeURIComponent(o))).protocol)throw new Error("Module Runtime: Must be HTTPS!");if(!/experience(-qa|-stage)?\.adobe\.com$/.test(o.hostname)&&!/localhost\.corp\.adobe\.com$/.test(o.hostname))throw new Error("Module Runtime: Invalid domain!");if(!/\.js$/.test(o.pathname))throw new Error("Module Runtime: Must be a JavaScript file!");t.EXC_US_HMR&&t.sessionStorage.setItem("unifiedShellMRScript",o.toString());var n=e.createElement("script");n.async=1,n.src=o.toString(),n.onload=n.onreadystatechange=function(){n.readyState&&!/loaded|complete/.test(n.readyState)||(n.onload=n.onreadystatechange=null,n=void 0,"EXC_MR_READY"in t&&t.EXC_MR_READY())},e.head.appendChild(n)})(document, window)
```
Comment (shazron) [inline generators/add-web-assets/exc-react/templates/src/utils.js:28]: wouldn't this script be better loaded off a file, for ease of update? unless there are bundler issues

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 1 above)
Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/App.js:59]: the purpose of the comment is to show how an example of how to use the handler, but yes in that case I should also comment out the conditional

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 2 above)
Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: same as above just showing an example, but yes we can argue that those comments might be confusing, are you in favor for deleting the commented code ?

Comment (shazron) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: yeah, delete. we've got source code control nowadays to view history 😅

Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: what do you mean by source code control? That comment was to show to the user how to use the handler in the context of the generated template :)
But yeah will delete

Comment (shazron) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: d'oh! forgot this was a generator.... 🤦

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 1 above)
Comment (shazron) [reply generators/add-web-assets/exc-react/templates/src/App.js:59]: d'oh! forgot this was a generator.... 🤦

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 2 above)
Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/App.js:48]: that's like inception code 😄

File: generators/add-web-assets/exc-react/templates/src/App.js (hunk 3 above)
Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/App.js:85]: oh good catch thanks!!

File: generators/add-web-assets/exc-react/templates/src/utils.js (hunk 4 above)
Comment (moritzraho) [reply generators/add-web-assets/exc-react/templates/src/utils.js:28]: no bundler issues, just wanted to limit the number of files the user sees.
but that's a fair point mentioned by @purplecabbage  too, so I guess that's a 2v1 😄 , will do put it back to the separate file

**Comment by moritzraho** [comment]
waiting on exc team to fix warnings

**Comment by moritzraho** [comment]
Fyi the source maps warnings have been fixed by the exc team. Thanks @shazron for the review!

@purplecabbage you had some concerns related to the history management, should we merge this or do you have some comments to add?

**Pull request merged**